src/checkout.js

```diff
@@ -0,0 +1,150 @@
+// bypasses the Checkout step to ask user for contributions
+import html from "../html/popup.html";
+import { centreElement, addAlbumToCart } from "./utilities.js";
+import Logger from "./logger";
+
+const albumsToPurchase = [1609998585];
+
+export default class Checkout {
+  constructor() {
+    this.log = new Logger();
+
+    this.config;
+    this.dialog;
+    this.checkout_button_sub;
+
+    this.checkoutButtonClicked = Checkout.checkoutButtonClicked.bind(this);
+    this.applyConfig = Checkout.applyConfig.bind(this);
+    this.closeDialogAndGoToCart = Checkout.closeDialogAndGoToCart.bind(this);
+    this.addAlbumToCart = addAlbumToCart.bind(this);
+
+    this.yesButtonClicked = Checkout.yesButtonClicked.bind(this);
+    this.noButtonClicked = Checkout.noButtonClicked.bind(this);
+
+    try {
+      this.port = chrome.runtime.connect(null, { name: "bandcamplabelview" });
+    } catch (e) {
+      if (e.message.includes("Error in invocation of runtime.connect")) {
+        this.log.error(e);
+        return;
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  init() {
+    this.log.info("Loaded Checkout");
+
+    this.checkout_button_sub = Checkout.replaceCheckoutButton();
+    this.checkout_button_sub.addEventListener(
+      "click",
+      this.checkoutButtonClicked
+    );
+
+    this.dialog = Checkout.createDialog();
+
+    this.dialog
+      .querySelector("#yes")
+      .addEventListener("click", this.yesButtonClicked);
+
+    this.dialog
+      .querySelector("#not_now")
+      .addEventListener("click", this.closeDialogAndGoToCart);
+
+    this.dialog
+      .querySelector("#no")
+      .addEventListener("click", this.noButtonClicked);
+
+    this.dialog.querySelector("#bes_close").addEventListener("click", () => {
+      this.dialog.style.display = "none";
+    });
+
+    this.port.onMessage.addListener(this.applyConfig);
+    this.port.postMessage({ requestConfig: {} }); // TO DO: this must be at end of init
+  }
+
+  static applyConfig(msg) {
+    this.log.info("config recieved from backend" + JSON.stringify(msg.config));
+    this.config = msg.config;
+  }
+
+  static checkoutButtonClicked() {
+    if (
+      !this.config.albumOnCheckoutDisabled &&
+      !this.config.albumPurchasedDuringCheckout
+    ) {
+      this.dialog.style.display = "block";
+      return;
+    }
+
+    this.closeDialogAndGoToCart();
+  }
+
+  static closeDialogAndGoToCart() {
+    this.dialog.style.display = "none";
+
+    const clickEvent = new MouseEvent("click", {
```

Comment:
unknown. Can you check with your tablet? I have no devices like this (and as far as I know extensions don't exist for mobile)

Reply:
They're `touch`, `touchstart` and `touchend` events. I forgot the platform target though, you're right this won't be used on mobile because it's an extension. Carry on